tests/integration/aliases/dotnet/adopt_into_component/step2/Program.cs

```diff
@@ -0,0 +1,114 @@
+﻿// Copyright 2016-2019, Pulumi Corporation.  All rights reserved.
+
+using System;
+using System.Threading.Tasks;
+using Pulumi;
+
+class Resource : ComponentResource
+{
+    public Resource(string name, ComponentResourceOptions options = null)
+        : base("my:module:Resource", name, options)
+    {
+    }
+}
+
+// Scenario #2 - adopt a resource into a component.  The component author is the same as the component user, and changes
+// the component to be able to adopt the resource that was previously defined separately...
+class Component : ComponentResource
+{
+    private Resource resource;
+
+    public Component(string name, ComponentResourceOptions options = null)
+        : base("my:module:Component", name, options)
+    { 
+        // The resource creation was moved from top level to inside the component.
+        this.resource = new Resource($"{name}-child", 
+            new ComponentResourceOptions
+            {
+                // With a new parent
+                Parent = this,
+                // But with an alias provided based on knowing where the resource existing before - in this case at top
+                // level.  We use an absolute URN instead of a relative `Alias` because we are referencing a fixed resource
+                // that was in some arbitrary other location in the hierarchy prior to being adopted into this component.
+                Aliases = { new Alias { Urn = "my:module:Resource::res2" } },
+            });
+    }
+}
+
+// Scenario 3: adopt this resource into a new parent.
+class Component2 : ComponentResource
+{
+    public Component2(string name, ComponentResourceOptions options = null) 
+        : base("my:module:Component2", name, options)
+    {        
+    }
+}
+
+
+// Scenario 4: Make a child resource that is parented by opts instead of 'this'.  Fix
+// in the next step to be parented by this.  Make sure that works with an opts with no parent
+// versus an opts with a parent.
+
+class Component3 : ComponentResource
+{
+    public Component3(string name, ComponentResourceOptions options = null) 
+        : base("my:module:Component3", name, options)
+    {
+        new Component2(name + "-child",
+            new ComponentResourceOptions 
+            { 
+                Aliases = { new Alias { Parent = options?.Parent } },
```

Comment:
this likely won't work.  if `options?.Parent` evaluates to null this effectively makes a do-nothing alias.  you'd have to write this as:

```new Alias { Parent = options?.Parent, NoParent = options?.Parent == null }```